### PR TITLE
Add assertion functions for parity with unittest

### PR DESCRIPTION
**Pull request by artorl**
Added assertion functions for parity with unittest (implemented as wrappers around unittest assertions).

Excluded:
- assertWarns, assertLogs, and assertWarnsRegex as
   they are implemented with their own context managers
   in unittest;
- assertDictContainsSubset (deprecated in unittest).

---
This change is [Reviewable](https://reviewable.io/reviews/google/mobly/769)

### CODE REVIEW

**Review by xpconanfan — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/769)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @artorl and @xpconanfan)

___
*[mobly/asserts.py, line 267 at r1](https://reviewable.io/reviews/google/mobly/769#-Mo1krKm7Tquc_hBRiho:-Mo1krKm7Tquc_hBRihp:bax1bm3) ([raw file](https://github.com/google/mobly/blob/419eb927c122b206d5ea31eb917ad5dc06ba0e73/mobly/asserts.py#L267)):*
> ```Python
> 
> 
> def assert_not_is_instance(obj, cls, msg=None, extras=None):
> ```

shouldn't this be "is not instance"?

**Comment by artorl** [comment]
> > _[Reviewable](https://reviewable.io/reviews/google/mobly/769)_ status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @artorl and @xpconanfan)
> 
> _[mobly/asserts.py, line 267 at r1](https://reviewable.io/reviews/google/mobly/769#-Mo1krKm7Tquc_hBRiho:-Mo1krKm7Tquc_hBRihp:bax1bm3) ([raw file](https://github.com/google/mobly/blob/419eb927c122b206d5ea31eb917ad5dc06ba0e73/mobly/asserts.py#L267)):_
> 
> > ```python
> > def assert_not_is_instance(obj, cls, msg=None, extras=None):
> > ```
> 
> shouldn't this be "is not instance"?

It's "assertNotIsInstance" in unittest, so I went with "assert_not_is_instance" for similarity.

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertNotIsInstance

**Comment by artorl** [comment]
I've made a change to fix unit tests on Python 3.6. Please approve workflow executions.

File: tests/mobly/asserts_test.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2016 Google Inc.
+# Copyright 2021 Google Inc.
```
Comment (xpconanfan) [inline tests/mobly/asserts_test.py:1]: this is actually meant to stay at the initial year.
same for the other one

**Review by xpconanfan — CHANGES_REQUESTED** [review]

File: tests/mobly/asserts_test.py (hunk 1 above)
Comment (artorl) [reply tests/mobly/asserts_test.py]: Done.

**Review by xpconanfan — APPROVED** [review]

**Review by eric100lin — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/769)* status: 0 of 2 files reviewed, 4 unresolved discussions (waiting on @artorl and @xpconanfan)

___
*[mobly/asserts.py, line 33 at r4](https://reviewable.io/reviews/google/mobly/769#-MoSOBWe5ejOnaL4acKc:-MoSOBWe5ejOnaL4acKd:buc5gsz) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/mobly/asserts.py#L33)):*
> ```
> Args:
> ```
nit: I think it's better to add pydoc for `**kwargs`:
`**kwargs: Keyword arguments to ...`.
___
*[mobly/asserts.py, line 55 at r4](https://reviewable.io/reviews/google/mobly/769#-MoSPCPlEQC3pYl9Nnm4:-MoSPCPlEQC3pYl9Nnm5:boh2z6g) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/mobly/asserts.py#L55)):*
> ```
> Assert
> ```
go/pystyle#function-docs
The docstring should be descriptive-style ("""Fetches rows from a Bigtable.""") rather than imperative-style ("""Fetch rows from a Bigtable.""").

**Review by eric100lin — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/769)* status: 0 of 2 files reviewed, 8 unresolved discussions (waiting on @artorl and @xpconanfan)

___
*[mobly/asserts.py, line 46 at r4](https://reviewable.io/reviews/google/mobly/769#-MoWXLfc0bptFqqkAqhJ:-MoWXLfc0bptFqqkAqhK:b-6wochj) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/mobly/asserts.py#L46)):*
> ```
> "%s %s"
> ```
Consider to use single quote(to align with strings in `class _AssertRaisesContext` in this file) and f-string here.
___
*[mobly/asserts.py, line 304 at r4](https://reviewable.io/reviews/google/mobly/769#-MoWYbJoAtRBbQdjgmWU:-MoWYbJoAtRBbQdjgmWV:b2eobf4) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/mobly/asserts.py#L304)):*
> ```
> Args
> ```
Missing the pydoc for `*args` and `**kwargs`.
___
*[tests/mobly/asserts_test.py, line 20 at r4](https://reviewable.io/reviews/google/mobly/769#-MoWb_We6xbLqZ2jJgcq:-MoWb_We6xbLqZ2jJgcr:bo2542r) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/tests/mobly/asserts_test.py#L20)):*
> ```
> "This is an expected exception."
> ```
Please consider to use single quote instead of double quote for strings in this file to align with rest of the files in mobly repo.
___
*[tests/mobly/asserts_test.py, line 26 at r4](https://reviewable.io/reviews/google/mobly/769#-MoWSzGSA4e1ap7L35Sr:-MoWSzGSA4e1ap7L35Ss:bk5brfk) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/tests/mobly/asserts_test.py#L26)):*
> ```
>   """Verifies that asserts.xxx functions raise the correct test signals.
>   """
> ```
Please move the three double-quote to the end of the summary line.

**Review by artorl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/769)* status: 0 of 2 files reviewed, 7 unresolved discussions (waiting on @artorl, @eric100lin, and @xpconanfan)

___
*[mobly/asserts.py, line 33 at r4](https://reviewable.io/reviews/google/mobly/769#-MoSOBWe5ejOnaL4acKc:-Moa7FKB81Ge01xYggrx:b-896fix) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/mobly/asserts.py#L33)):*
<details><summary><i>Previously, eric100lin (Eric Lin (Tzu Hsiang Lin)) wrote…</i></summary><blockquote>

> ```
> Args:
> ```
nit: I think it's better to add pydoc for `**kwargs`:
`**kwargs: Keyword arguments to ...`.
</blockquote></details>

Done.
___
*[mobly/asserts.py, line 46 at r4](https://reviewable.io/reviews/google/mobly/769#-MoWXLfc0bptFqqkAqhJ:-Moa9aTh9808PxiSL5EP:b-896fix) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/mobly/asserts.py#L46)):*
<details><summary><i>Previously, eric100lin (Eric Lin (Tzu Hsiang Lin)) wrote…</i></summary><blockquote>

> ```
> "%s %s"
> ```
Consider to use single quote(to align with strings in `class _AssertRaisesContext` in this file) and f-string here.
</blockquote></details>

Done.
___
*[mobly/asserts.py, line 55 at r4](https://reviewable.io/reviews/google/mobly/769#-MoSPCPlEQC3pYl9Nnm4:-Moa7rGu4_lJqghxoFPk:b82bsz3) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/mobly/asserts.py#L55)):*
<details><summary><i>Previously, eric100lin (Eric Lin (Tzu Hsiang Lin)) wrote…</i></summary><blockquote>

> ```
> Assert
> ```
go/pystyle#function-docs
The docstring should be descriptive-style ("""Fetches rows from a Bigtable.""") rather than imperative-style ("""Fetch rows from a Bigtable.""").
</blockquote></details>

Done for all new assertions.

Although this is inconsistent with the existing assertions in asserts.py, which use imperative-style docstrings.
___
*[mobly/asserts.py, line 304 at r4](https://reviewable.io/reviews/google/mobly/769#-MoWYbJoAtRBbQdjgmWU:-Moa6OFMAso20lfGxDwX:bwh7emz) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/mobly/asserts.py#L304)):*
<details><summary><i>Previously, eric100lin (Eric Lin (Tzu Hsiang Lin)) wrote…</i></summary><blockquote>

> ```
> Args
> ```
Missing the pydoc for `*args` and `**kwargs`.
</blockquote></details>

I haven't modified this docstring.

I'm not sure why both assert_raises and assert_raises_regex accept *args and **kwargs but don't use them. Is this intentional?
___
*[tests/mobly/asserts_test.py, line 20 at r4](https://reviewable.io/reviews/google/mobly/769#-MoWb_We6xbLqZ2jJgcq:-Moa96zgBLf17ZHyJ6AM:b-896fix) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/tests/mobly/asserts_test.py#L20)):*
<details><summary><i>Previously, eric100lin (Eric Lin (Tzu Hsiang Lin)) wrote…</i></summary><blockquote>

> ```
> "This is an expected exception."
> ```
Please consider to use single quote instead of double quote for strings in this file to align with rest of the files in mobly repo.
</blockquote></details>

Done.
___
*[tests/mobly/asserts_test.py, line 26 at r4](https://reviewable.io/reviews/google/mobly/769#-MoWSzGSA4e1ap7L35Sr:-Moa8GOY9zA1JhciRjrm:b-896fix) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/tests/mobly/asserts_test.py#L26)):*
<details><summary><i>Previously, eric100lin (Eric Lin (Tzu Hsiang Lin)) wrote…</i></summary><blockquote>

> ```
>   """Verifies that asserts.xxx functions raise the correct test signals.
>   """
> ```
Please move the three double-quote to the end of the summary line.
</blockquote></details>

Done.

**Review by xpconanfan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/769)* status: 0 of 2 files reviewed, 7 unresolved discussions (waiting on @eric100lin and @xpconanfan)

___
*[mobly/asserts.py, line 304 at r4](https://reviewable.io/reviews/google/mobly/769#-MoWYbJoAtRBbQdjgmWU:-MoaKYei5vUfu0t9dlPU:b-4mpfrg) ([raw file](https://github.com/google/mobly/blob/98fa3c165bb2c9fcd708ed3a46f37a45ab273b96/mobly/asserts.py#L304)):*
<details><summary><i>Previously, artorl (Art Orlov) wrote…</i></summary><blockquote>

I haven't modified this docstring.

I'm not sure why both assert_raises and assert_raises_regex accept *args and **kwargs but don't use them. Is this intentional?
</blockquote></details>

I don't remember either...
I'd assume this is an error?

Let's leave them as is for now.
We can remove in a separate PR

**Review by eric100lin — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/google/mobly/769#-:-MoaaQZlDvPO_ji52AOO:bnfp4nl)* status: 0 of 2 files reviewed, 7 unresolved discussions (waiting on @artorl, @eric100lin, and @xpconanfan)

**Review by eric100lin — COMMENTED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/mobly/769)* status: all files reviewed, 3 unresolved discussions (waiting on @eric100lin and @xpconanfan)

**Pull request merged**